jwst/transforms/models.py

```diff
@@ -10,24 +9,129 @@
 from astropy.modeling.models import Polynomial2D
 
 
-__all__ = ['AngleFromGratingEquation', 'WavelengthFromGratingEquation', 'NRSZCoord',
-           'Unitless2DirCos', 'DirCos2Unitless', 'Rotation3DToGWA', 'Gwa2Slit', 'Slit2Msa',
-           'slitid_to_slit', 'slit_to_slitid']
+__all__ = ['AngleFromGratingEquation', 'WavelengthFromGratingEquation',
+           'NRSZCoord', 'Unitless2DirCos', 'DirCos2Unitless',
+           'Rotation3DToGWA', 'Gwa2Slit', 'Slit2Msa', 'slitid_to_slit',
+           'slit_to_slitid', 'Snell', 'RefractionIndex']
 
 
 # Number of shutters per quadrant
 N_SHUTTERS_QUADRANT = 62415
 
 
+class RefractionIndex(Model):
+    """Calculate a refraction index given lamda and a reference file.
+
+    Referenced in Muzzerole's IDL program
+    Requires  reference file information
+
+    lam is in angstroms
+    temp is in kelvin
+    pressure is in atmospheres
+    pref is the reference pressure
+    kcoef is a 3 item list
+    lcoef is a 3 item list
+    tref is the reference temperature
+    """
+
+    fittable = False
+    separable = False
+
+    inputs = ("lam",)
+    outputs = ("n",)
+
+    temp = Parameter()
+    tref = Parameter()
+    pref = Parameter
+    pressure = Parameter()
+    kcoef = Parameter()
+    lcoef = Parameter()
+    tcoef = Parameter()
+
+    def __init__(self, lam, temp, tref, pref, pressure, kcoef, lcoef, tcoef,
```

Comment:
`lam` is an input and is not used to initialize the model, take it out of the `__init__`.